ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -0,0 +1,284 @@
+import org.opensearch.ml.common.transport.deploy.MLDeployModelResponse;
+import org.opensearch.ml.common.transport.model.MLModelGetAction;
+import org.opensearch.ml.common.transport.model.MLModelGetRequest;
+import org.opensearch.ml.common.transport.model.MLModelGetResponse;
+import org.opensearch.ml.common.transport.model.MLModelSearchAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput;
+import org.opensearch.ml.common.transport.register.MLRegisterModelRequest;
+import org.opensearch.ml.common.transport.register.MLRegisterModelResponse;
+import org.opensearch.ml.common.transport.task.MLTaskGetAction;
+import org.opensearch.ml.common.transport.task.MLTaskGetRequest;
+import org.opensearch.ml.common.transport.task.MLTaskGetResponse;
+import org.opensearch.ml.engine.ExecuteException;
+import org.opensearch.ml.engine.algorithms.DLModelExecute;
+import org.opensearch.ml.engine.annotation.Function;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BooleanSupplier;
+
+import static org.opensearch.index.query.QueryBuilders.termQuery;
+
+@Log4j2
+@NoArgsConstructor
+@Function(FunctionName.METRICS_CORRELATION)
+public class MetricsCorrelation extends DLModelExecute {
+
+    private static final int AWAIT_BUSY_THRESHOLD = 1000;
+    private Client client;
+    private Settings settings;
+
+    private static final String MODEL_NAME = "METRICS_CORRELATION";
+    public static final String MCORR_ML_VERSION = "1.0.0";
+
+    //TODO: Model didn't publish yet to the release repo, so using this for the development.
+    // But before merging this code, we will have the release artifact url here for
+    public static final String MCORR_MODEL_URL =
+            "https://github.com/dhrubo-os/semantic-os/raw/main/mcorr.zip";
+
+    public MetricsCorrelation(Client client, Settings settings) {
+        this.client = client;
+        this.settings = settings;
+    }
+
+    @Override
+    public MetricsCorrelationOutput execute(Input input) throws ExecuteException {
+        if (!(input instanceof MetricsCorrelationInput)) {
+            throw new IllegalArgumentException("wrong input");
+        }
+        List<MCorrModelTensors> tensorOutputs = new ArrayList<>();
+        MetricsCorrelationInput metricsCorrelation = (MetricsCorrelationInput) input;
+        List<float[]> inputData = metricsCorrelation.getInputData();
+
+        float[][] processedInputData = processedInput(inputData);
+
+        AtomicReference<String> loadedModelId = new AtomicReference<>();
+
+        SearchRequest modelSearchRequest = getSearchRequest();
+        searchModel(modelSearchRequest, ActionListener.wrap(modelInfo -> {
```

Comment:
How about make `loadedModelId` as field of this class? If `loadedModelId` has valid model id, then we don't need to search model, register and deploy.